src/domain/products/details/variants/variant-editor.tsx

```diff
@@ -23,6 +24,12 @@ const VariantEditor = ({ variant, onSubmit, onCancel }) => {
   const { store, isLoading } = useAdminStore()
   const [currencyOptions, setCurrencyOptions] = useState([])
   const [prices, setPrices] = useState(variant.prices)
+  const [metadata, setMetadata] = useState(
+    Object.keys(variant.metadata).map((key) => ({
+      key,
+      value: variant.metadata[key],
+    }))
+  )
```

Comment:
Why not just:
```suggestion
  const [metadata, setMetadata] = useState(variant.metadata)
```

Reply:
@srindom the metadata form wants a shape of `{key: string, value: string}[]`, but the metadata is just type `{[x:string]: string}`

Reply:
@srindom the metadata form wants a shape of `{key: string, value: string}[]`, but the metadata is just type `{[x:string]: string}`